### PR TITLE
fix(ios): instead of ignoring call give error in case of nil url passed to Ti.Network.HTTPClient

### PR DESCRIPTION
**Pull request by vijaysingh-axway**
https://jira.appcelerator.org/browse/TIMOB-27767

### CODE REVIEW

File: tests/Resources/ti.network.httpclient.addontest.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+describe('Titanium.Network.HTTPClient', function () {
+	this.timeout(6e4);
+
+	it('TIMOB-27767 - trigger error callback for invalid URL', function (finish) {
+		var xhr = Ti.Network.createHTTPClient();
+		xhr.setTimeout(6e4);
+
+		xhr.onerror = function (e) {
+			finish();
+		};
```
Comment (sgtcoolguy) [inline tests/Resources/ti.network.httpclient.addontest.js:21]: add `onload` handler and quickly fail the test if it gets fired?
```js
xhr.onload = e => finish(new Error('Expected onerror to be triggered for invalid URL'));
```

**Review by sgtcoolguy — CHANGES_REQUESTED** [review]
It's not clear to me where the actual fix/implementation is here. I just see the test case...

File: tests/Resources/ti.network.httpclient.addontest.js (hunk 1 above)
Comment (vijaysingh-axway) [reply tests/Resources/ti.network.httpclient.addontest.js:21]: @sgtcoolguy  Actual fix is in APSHTTPClient library . See PR https://github.com/appcelerator/APSHTTPClient/pull/50 .

In jira ticket there are 3 PRs.
1. https://github.com/appcelerator/APSHTTPClient/pull/50 -  Actual fix
2. https://github.com/appcelerator/aps_sdk/pull/341 - To use updated libAPSHTTPClient.a
3. https://github.com/appcelerator/titanium_mobile/pull/11513 - To use updated libraries in SDK.

Comment (vijaysingh-axway) [reply tests/Resources/ti.network.httpclient.addontest.js:21]: 'onload' event was not fired in this case. I think there is no need to add 'onload' handler.

**Review by sgtcoolguy — APPROVED** [review]

**Comment by ssekhri** [comment]
FR Passed
Verified on:
Mac OS: 10.15.4
SDK: 9.1.0.v20200701133108
Appc CLI: 8.0.0
JDK: 11.0.6
Node: 10.17.0
Studio: 6.0.0.202005141803
Xcode: 11.4

**Pull request merged**